### ISSUE TITLE
Ember-CLI 2.15.0-beta.1 and greater pass autoboot options to browser JS

### ISSUE THREAD

**Issue by mixonic** [ember-fastboot/ember-cli-fastboot#502]
**tl;dr** Ember 2.15+ apps do not boot client-side application code with ember-cli-fastboot.

With Ember-CLI 2.14, ember-cli-fastboot will generate `{{app-boot}}` content like the following:

```js
if (typeof FastBoot === 'undefined') {
  if (!runningTests) {
    require('created/app')['default'].create({"name":"created","version":"0.0.0+60164923"});
  }
}

define('~fastboot/app-factory', ['created/app', 'created/config/environment'], function(App, config) {
  App = App['default'];
  config = config['default'];

  return {
    'default': function() {
      return App.create(config.APP);
    }
  };
});
```

In Ember-CLI 2.15.0-beta.1 and greater, the `{{CONFIG_APP}}` value that becomes arguments to `create` in the browser ([source](https://github.com/ember-fastboot/ember-cli-fastboot/blob/master/lib/utilities/fastboot-app-module.js#L15)) now includes `"autoboot": false`:

```js
if (typeof FastBoot === 'undefined') {
  if (!runningTests) {
    require('created/app')['default'].create({"name":"created","version":"0.0.0+60164923","autoboot":false});
  }
}

define('~fastboot/app-factory', ['created/app', 'created/config/environment'], function(App, config) {
  App = App['default'];
  config = config['default'];

  return {
    'default': function() {
      return App.create(config.APP);
    }
  };
});
```

This prevents the browser from booting the app. It isn't completely clear if Ember-CLI fixed a bad bug ember-cli-fastboot was reliant upon, if Ember-CLI introduced a regression, or if there is something less than optimal on both sides.

/cc @kratiahuja @rwjblue

**Comment by rwjblue** [comment]
Thank you for reporting with such detail. 

It seems this is caused by a recent addition of caching the output of `project.confjg()` calls. Previously, every time you call `project.config()` you would get a fresh object and could therefore treat it as your own "copy". We determined that building the config is actually fairly expensive so we added caching, which ultimately meant that this mutation we were doing now modifies the single shared config. 

@kratiahuja has a fix up in #503 for this.

**Comment by mixonic** [comment]
Closing this in favor of https://github.com/ember-fastboot/ember-cli-fastboot/pull/503, which I can confirm fixes the issue. Thank you!
